django/db/backends/base/base.py

```diff
@@ -633,6 +633,7 @@ def schema_editor(self, *args, **kwargs):
 
     def on_commit(self, func):
         if self.in_atomic_block:
+            assert callable(func), 'on_commit() callback must be callable.'
```

Comment:
I would raise a `TypeError`, e.g.:
```suggestion
            if not callable(func):
                raise TypeError("on_commit()'s callback must be a callable.")
```